### ISSUE TITLE
Receive Addresses Missing

### ISSUE THREAD

**Issue by MyS5Droid** [spesmilo/electrum#3855]
Hi,

I have Electrum 3.0.5 (Wincrap)

I logged in to wallet today to find ALL receive addresses missing... I did a test, and the addresses are still owned by the wallet but they just do not show in the receive address page...

I want to reuse some addresses for some things...

Please fix ASAP

Thanks

**Comment by SomberNight** [comment]
> I logged in to wallet today to find ALL receive addresses missing

> they just do not show in the receive address page

What exactly do you mean by this? In which tab are you looking for the addresses?

> I did a test, and the addresses are still owned by the wallet

What test?

**Comment by AbdussamadA** [comment]
You can view all your addresses on the addresses tab. Go to view  menu > show addresses to see that tab. You can search for addresses by pressing ctrl+f and fiddling with the filters up top.

**Comment by MyS5Droid** [comment]
Yes that is what I mean the labels are all gone. I was looking at receiving tab... and checking in console. they still exist but they all disappeared and the labels are gone.

"never expire" I would think means never expires, as in never goes away...

**Comment by bauerj** [comment]
Did you receive transactions on that addresses?

**Comment by ecdsa** [comment]
I guess you pressed on the filter button, and are looking at your change addresses

**Comment by MyS5Droid** [comment]
No I did not receive anything... I am "waiting" for them and want to see them there until they are used so I don't have to regenerate and relabl them all again.

No the filter button is not pressed.

I have since moved to bitcoin full node.

The problem still exists.

Thanks

**Comment by ecdsa** [comment]
you are too unclear about what the issue is.
please post a screenshot.

**Comment by MyS5Droid** [comment]
As I said, I moved to Bitcoin Full Node in Debian 9... I no longer use Electrum Windows anymore.


I am not sure if I am speaking proper English... It's supposed to be my native language.

I said basically I add receive addresses with descriptions, and then after closing and opening electrum they all disappear - even though I set them to never expire.

Not sure why you need screenshots of addresses which are not there to screenshot in the first place...

1. Click on "Receive" tab
2. Click on "New" to generate new address
3. Add description
4. Click expiry dropdown, select "never"
5. Go back to step 1 - repeat for 5 or so addresses
6. Close Electrum
7. Open Electrum
8. Click "Receive" tab
9. Notice the addresses missing
10. Optionally go to the console, and check the address, it still owned by the wallet but the entry on the "Receive" tab is gone along with the description telling you what the address is for

I am done reporting the bug, I have moved off to BTC full node and I am no longer able to debug or perform tests.

However if addresses are just disappearing this should be fixed. If I were a noob (I'm not) I would have no idea that the wallet still owns the address and would be freaking out about my pending transfers whether I will receive them or not.

Have a good day

**Comment by SomberNight** [comment]
Thanks for the description.

My only guess atm is that the storage is not written back to disk when you close Electrum.
(after you "save" receive requests, they only get saved into memory; and will get persisted to disk during the next `storage.write()`)
However when you close Electrum, normally the storage is written back to disk.

I could only reproduce by killing the process.

**Comment by alexwo** [comment]
It seems that this issue still reproduces on 3.3.8 OSX version of electrum 

Steps to reproduce:

1. Create a new receiving request
2. KIll the electrum process or reboot your mac
3. Startup the electrum app

**Expected result**
The newly create receiving request is listed in the receive tab

**Actual result**
1. The receiving request does not exist after the restart in the receiving tab
2. It is still possible to find it still  in the wallet address tab but without the original label
3. As a workaround, it is possible to click on the receiving address and select request a payment, this will allow you to attach a new label and save the address in the receive tab. 

I actually requested a transaction to such requesting address that disappeared :)
hope the transaction will Pass successfully in the end. I guess it should as the receiving address was still stored in the wallet Addresses tab.

**Comment by ecdsa** [comment]
yes, a bitcoin address is permanently valid
